# Level 15: whom?

[http://www.pythonchallenge.com/pc/return/uzi.html](http://www.pythonchallenge.com/pc/return/uzi.html)

## 線索

一張一月份月曆的照片，年份的部分被破壞了，看不出來是 1X6 年還是 1XX6 年，1/26 的部分被圈了起來

這一年的 1/26 是星期一，右下角有前後各一個月份的縮圖，看得出來二月份有 29 天，這年是閏年

看看 html source，有兩句註解，應該是提示

> he ain't the youngest, he is the second

> todo: buy flowers for tomorrow

感覺應該跟圈起來的日期有關

## 嘗試

先找出所有 1X6 跟 1XX6 的年份，過濾出該年是閏年並且 1/26 是星期一的年份

In [1]:
from calendar import isleap, weekday

years = [i for i in range(106, 197) if i % 10 == 6] + [i for i in range(1006, 1997) if i % 10 == 6]

leap_years = [y for y in years if isleap(y)]
print(leap_years)

[116, 136, 156, 176, 196, 1016, 1036, 1056, 1076, 1096, 1116, 1136, 1156, 1176, 1196, 1216, 1236, 1256, 1276, 1296, 1316, 1336, 1356, 1376, 1396, 1416, 1436, 1456, 1476, 1496, 1516, 1536, 1556, 1576, 1596, 1616, 1636, 1656, 1676, 1696, 1716, 1736, 1756, 1776, 1796, 1816, 1836, 1856, 1876, 1896, 1916, 1936, 1956, 1976, 1996]


In [2]:
# weekday() return 0~6 as Mon~Sun
matched_years = [y for y in leap_years if weekday(y, 1, 26) == 0]
print(matched_years)

[156, 1176, 1356, 1576, 1756, 1976]


由第一句的註解提示，應該是要找第二年輕的年份？那麼應該就是 1756 年

第二句的 todo 說要為明天買花，那個應該是指 1/26 的待辦事項

[所以我們來 google 看看 1756/1/27 有什麼相關的訊息](https://www.google.com/search?q=1756/1/27)

噢，看起來應該是跟莫札特有關 (Wolfgang Amadeus Mozart)

試試看 ```http://www.pythonchallenge.com/pc/return/mozart.html``` 就順利進入下一關囉